middleware/errorcounter.go

```diff
@@ -30,23 +36,35 @@ func ErrorCounterWithConfig(config ErrorCounterConfig) gorabbitmq.MiddlewareFunc
 			err := hf(c)
 
 			if err != nil {
-				c.Ack(false)
 				table := c.Delivery().Headers
 				if table == nil {
 					table = make(map[string]interface{})
 				}
 
 				if errorCounter, ok := c.Value(ErrorCounter{}).(int32); ok {
 					if errorCounter >= config.MaxRetries {
-						config.MaxRetriesExceeded(c)
+						retryError := config.MaxRetriesExceeded(c)
+						if retryError != nil {
+							return errors.Wrap(retryError, fmt.Sprintf("failed to execute maxRetriesExceeded handler, after %v", err))
```

Comment:
Please use the `Wrapf` function, provided by the `"github.com/pkg/errors"` package, so we dont have to import the `"fmt"` package.

Reply:
fixed with 6fb874b08be0e80f9026fc22c75d7c7b0503be72